spec/controllers/authenticate_controller_spec.rb

```diff
@@ -6,6 +6,47 @@
   let(:password) { "password" }
   let(:login) { "u-#{random_hex}" }
   let(:account) { "rspec" }
+  let(:authenticator) { "authn" }
+
+  context "#login" do
+    shared_examples_for "successful login" do
+      it "succeeds" do
+        post :login, params
+        expect(response).to be_ok
+        expect(response.body).to eq(api_key)
+      end
+    end
+    
+    shared_examples_for "login denied" do
+      it "is unauthorized" do
+        post :login, account: account, authenticator: authenticator
+        expect(response.code).to eq("401")
+      end
+    end
+    
+    context "without auth" do
+      it_should_behave_like "login denied"
+    end
+
+    context "when user doesn't exist" do
+      let(:basic_password) { "the-password" }
+      include_context "authenticate Basic"
+      it_should_behave_like "login denied"
+    end
+
+    context "when user exists" do
+      include_context "create user"
+      context "with basic auth" do
+        let(:basic_password) { api_key }
+        include_context "authenticate Basic"
+        it_should_behave_like "successful login"
+      end
+      context "with Token auth" do
+        include_context "authenticate Token"
+        it_should_behave_like "login denied"
+      end
+    end
+  end
```

Comment:
One of the main reason for creating `Strategy` as an object decoupled is so that we can test this logic in pure ruby, without going through Rails.  The same tests, written as unit tests, will execute instantly, as well as be more focused (ie, you're just testing input / output of a method).